be/src/exec/pipeline/pipeline_driver_dispatcher.cpp

```diff
@@ -107,7 +139,11 @@ void GlobalDriverDispatcher::run() {
             switch (driver_state) {
             case READY:
             case RUNNING: {
-                this->_driver_queue->put_back_from_dispatcher(driver);
+                if constexpr (is_use_workgroup) {
+                    this->_driver_queue->put_back_from_dispatcher(driver);
+                } else {
+                    this->_driver_queue->put_back(driver);
+                }
```

Comment:
we could make `QuerySharedDriverQueue::put_back_from_dispatcher()` call `QuerySharedDriverQueue::put_back()`, and then we needn't judge whether `is_use_workgroup` here.